docs/source/cheat_sheet.rst

```diff
@@ -11,121 +11,158 @@ language represents various common types in Python 2.
    x = [1] # type: List[int]
    x = set([6, 7]) # type: Set[int]
-   # For mappings, we need the types of both keys and values
+
+   # For mappings, we need the types of both keys and values.
    x = dict(field=2.0) # type: Dict[str, float]
-   # For tuples, we specify the types of all the elements
+
+   # For tuples, we specify the types of all the elements.
    x = (3, "yes", 7.5) # type: Tuple[int, str, float]
 
-   # six.text_type is str/unicode in Python 2 and string but not bytes in Python 3
-   x = ["string", u"unicode"] # type: List[text_type]
-   # If something could be one of a few types, use Union
-   x = [3, 5, "test", "fun"] # type: List[Union[int, str]]
-   x = re.match(r'[0-9]+', "15") # type: Match[str]
+   # For textual data, we generally use six.text_type.
+   # This is `unicode` in Python 2 and `str` in Python 3.
+   x = ["string", u"unicode"] # type: List[six.text_type]
 
-   # If you don't know the type of something, you can use Any
-   x = mystery_function() # type: Any
-   # And if you want to have something not be type-checked, you can
-   # use ignore to suppress mypy warnings for a given line
-   # Ideally, one would never use this
-   x = confusing_function() # type: ignore
+   # Use Optional for values that could be None.
+   input_str = f() # type: Optional[str]
+   if input_str is not None:
+      print input_str
 
-   # This is how you annotate a function definition
+
+Functions
+*********
+
+.. code-block:: python
+
+   from typing import Callable, Iterable
+
+   # This is how you annotate a function definition.
    def stringify(num):
        # type: (int) -> str
        """Your function docstring goes here after the type definition."""
        return str(num)
 
-   # And here's how you specify multiple arguments
+   # And here's how you specify multiple arguments.
    def plus(num1, num2):
        # type: (int, int) -> int
        return num1 + num2
 
-   # Add type annotations for kwargs as though they were positional args
+   # Add type annotations for kwargs as though they were positional args.
    def f(num1, my_float=3.5):
        # type: (int, float) -> float
        return num1 + my_float
-   # This is how you annotate a function value
+
+   # This is how you annotate a function value.
    x = f # type: Callable[[int, float], float]
 
-   # Use Optional[Type] for objects that could be None
-   def f(input_str=None):
-       # type: (Optional[str]) -> int
-       if input_str is not None:
-           return len(input_str)
-       return 0
-
-   from typing import Mapping, MutableMapping
-   # Dict is a python dictionary
-   # MutableMapping is an abstract base class for a dict-type thing
-   # Mapping is an abtract base class for a dict-type thing that may
-   # not support writing to the mapping
+   # A generator function that yields ints is secretly just a function that
+   # returns an iterable (see below) of ints, so that's how we annotate it.
```

Comment:
Yeah. I wrote Iterator first for that reason, then contemplated explaining the distinction between Iterator and Iterable (having already recommended Iterable elsewhere), and concluded there isn't a good explanation in this context because the distinction doesn't actually matter.